SortingAlgorithms/Interfaces/ISortingAlgorithm.cs

```diff
@@ -8,6 +8,6 @@ namespace SortingAlgorithms.Interfaces
 {
     interface ISortingAlgorithm
     {
-        void Sort(int[] array);
+        void Sort<T>(T[] array) where T : IComparable;
```

Comment:
Please, add xml comments to all methods. You can use <inheritdoc/> for inheritors.
For example, here, method Sort should be filled xml comments and BinarySearch and LinearSearch can be marked as <inheritdoc/>.